publisher/publisher.go

```diff
@@ -448,17 +453,33 @@ func uploadRpm(conf config, srcTemplate string, upload Upload, arch string) (err
 			_ = os.Remove(signaturePath)
 		}
 
-		// "cache" the repodata so it doesnt have to process all again
-		if err = execLogOutput(l, "cp", "-rf", repoPath+"/repodata/", os.TempDir()+"/repodata/"); err != nil {
-			return err
+		// check for .repo file
+		if _, err = os.Stat(repoFilePath); conf.accessPointHost != "" && os.IsNotExist(err) {
+			l.Println(fmt.Sprintf("creating 'newrelic-infra.repo' file in %s", repoPath))
+
+			repoFileContent := fmt.Sprintf(
+				"[newrelic-infra]\nname=New Relic Infrastructure\nbaseurl=%s/%s\ngpgkey=https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg\ngpgcheck=1\nrepo_gpgcheck=1",
```

Comment:
Perhaps we could use ` literals rather than escaping newlines for better readability